consul_registry.go

```diff
@@ -81,16 +89,26 @@ func (c *consulRegistry) Register(info *registry.Info) error {
 	if err != nil {
 		return err
 	}
+
 	svcID, err := getServiceId(info)
 	if err != nil {
 		return err
 	}
+
+	var svcTags []string
+	for k, v := range info.Tags {
+		if strings.Contains(k, kvJoinChar) || strings.Contains(v, kvJoinChar) {
+			return errillegalTagChar
+		}
+		svcTags = append(svcTags, fmt.Sprintf("%s%s%s", k, kvJoinChar, v))
+	}
```

Comment:
1. Better refactor this into a new func with unit tests.
2. slice can be initialized with capacity = len(info.Tags)

Reply:
And maybe we can weaken the restriction only to keys, as long as we separator "k:v" to only 2 parts.

Reply:
> And maybe we can weaken the restriction only to keys, as long as we separator "k:v" to only 2 parts.

I've thought about it. But I check key illegal char to avoid some bugs. 
For example:
```
registry.Info.Tags = map[string]string{"we:are":"ateam"}
```
Register function, conv it to `[]string{"we:are:ateam"}`
then,Resolve function,how to split it？

Reply:
As a replied in the following conversation, the restriction is only necessary for keys, with which we can always separate "`key`:`value`" into exactly 2 segments, regardless whether values contain ":"